src/v/pandaproxy/schema_registry/errors.h

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "outcome.h"
+#include "pandaproxy/schema_registry/error.h"
+#include "pandaproxy/schema_registry/exceptions.h"
+#include "pandaproxy/schema_registry/types.h"
+
+#include <fmt/format.h>
+
+#include <type_traits>
+
+namespace pandaproxy::schema_registry {
+
+class error_info {
+public:
+    error_info() = default;
+    error_info(error_code ec, std::string msg)
+      : _ec{ec}
+      , _msg{std::move(msg)} {}
+
+    const error_code& code() const noexcept { return _ec; }
+    const std::string& message() const noexcept { return _msg; }
+
+private:
+    error_code _ec;
+    std::string _msg;
+};
+
+inline exception as_exception(const error_info& ei) {
```

Comment:
this `exception` is hard to read. i think we should rename `sr_exception`

Reply:
> this `exception` is hard to read. i think we should rename `sr_exception`

It's namespaced, but I understand. I have some plans to make this more general, along with the interop with `result<T>` and coroutines. Not quite nailed it yet, but I'm approaching something nice.